core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -194,7 +195,9 @@ public final void commit(DataContainer container, EntryVersion newVersion) {
             if (isRemoved() && !isEvicted()) ahm.markRemoved(true);
          }
 
-         if (isRemoved()) {
+         if (isEvicted() && container instanceof DefaultDataContainer) {
```

Comment:
Talked with Horia and they don't use the DataContainer at all so this should be fine as is.